flaky mDNS test: TestOtherDiscovery

```
=== RUN   TestOtherDiscovery
      mdns_test.go:99: 
          	Error Trace:	mdns_test.go:99
          	Error:      	Condition never satisfied
          	Test:       	TestOtherDiscovery
          	Messages:   	expected peers to find each other
  --- FAIL: TestOtherDiscovery (26.31s)

--- a/go.mod
+++ b/go.mod
@@ -41,8 +41,7 @@ require (
 	github.com/libp2p/go-stream-muxer-multistream v0.3.0
 	github.com/libp2p/go-tcp-transport v0.3.0
 	github.com/libp2p/go-ws-transport v0.5.0
-	github.com/libp2p/zeroconf/v2 v2.0.0
-	github.com/miekg/dns v1.1.43 // indirect
+	github.com/libp2p/zeroconf/v2 v2.1.1
 	github.com/multiformats/go-multiaddr v0.4.0
 	github.com/multiformats/go-multiaddr-dns v0.3.1
 	github.com/multiformats/go-multistream v0.2.2

